test/direct.js

```diff
@@ -0,0 +1,115 @@
+
+
+describe('Direct upload of ERC', function () {
+    describe('POST /api/v1/compendium response with executable ERC', () => {
+        before((done) => {
+            let req = createCompendiumPostRequest(host, './test/erc/executable', cookie_o2r);
+
+            request(req, (err, res, body) => {
+                assert.ifError(err);
+                done();
+            });
+        });
+
+        it('should respond with HTTP 200 OK', (done) => {
+            request(host + '/api/v1/compendium', (err, res, body) => {
+                let req = createCompendiumPostRequest(host, './test/erc/executable', cookie_o2r);
+
+                request(req, (err, res, body) => {
+                    assert.ifError(err);
+                    assert.equal(res.statusCode, 200);
+                    done();
+                });
+            });
+        });
+
+        it('should respond with valid JSON', (done) => {
+            request(host + '/api/v1/compendium', (err, res, body) => {
+                let req = createCompendiumPostRequest(host, './test/erc/executable', cookie_o2r);
+
+                request(req, (err, res, body) => {
+                    assert.ifError(err);
+                    assert.isObject(JSON.parse(body), 'returned JSON');
+                    done();
+                });
+            });
+        });
+
+        it('should give a response including the id field', (done) => {
+            request(host + '/api/v1/compendium', (err, res, body) => {
+                let req = createCompendiumPostRequest(host, './test/erc/executable', cookie_o2r);
+
+                request(req, (err, res, body) => {
+                    assert.ifError(err);
+                    assert.isDefined(JSON.parse(body).id, 'returned id');
+                    assert.property(JSON.parse(body), 'id');
+                    done();
+                });
+            });
+        });
+    });
+
+
+    describe('POST /api/v1/compendium with invalid bag', () => {
+        it('should fail the upload because bag is invalid', (done) => {
+            let req = createCompendiumPostRequest(host, './test/erc/invalid_bag', cookie_o2r);
+
+            request(req, (err, res, body) => {
+                assert.ifError(err);
+                assert.equal(res.statusCode, 400);
+                assert.include(body, 'bag ist invalid');
+                done();
+            });
+        }).timeout(requestLoadingTimeout);
+
+        it('should not tell about internal server configuration in the error message', (done) => {
+            let req = createCompendiumPostRequest(host, './test/erc/invalid_bag', cookie_o2r);
+
+            request(req, (err, res, body) => {
+                assert.ifError(err);
+                assert.notInclude(body, config.fs.base);
+                done();
+            });
+        }).timeout(requestLoadingTimeout);
+    });
+
+    describe.skip('POST /api/v1/compendium with virus', () => {
```

Comment:
Yes. And it must be skipped because we're not testing that yet. I'd like to leave it in as a reminder to take another try at that feature later, see https://github.com/o2r-project/o2r-muncher/issues/39